clang/test/CXX/drs/dr158.cpp

```diff
@@ -1,7 +1,7 @@
-// RUN: %clang_cc1 -triple x86_64-linux -std=c++98 %s -O3 -disable-llvm-passes -pedantic-errors -emit-llvm -o - | FileCheck %s
```

Comment:
Can you explain why `-disable-llvm-passes` is there, and why is can be removed now?

Reply:
Yes,

The matter is that if no llvm passes are run, IR is unoptimized and contains extra load and store instruction that operate with stack (storing and loading params there). Those extra loads should not be considered by the test, because it's goal to check tbaa for specific load and store coming from line 22 and 24. Before I introduced more precise metadata on pointer types, all of that load and stores had same metadata, so test passed even if wrong load was considered. Now, this test should pick only load coming from line 22. So, to make things easier, I just enabled optimization passes and final IR contains only one load and one store instruction.

Reply:
This is okay as long as you're sure the optimization will be performed and you make sure your test only checks the pattern you want out of the optimizer and doesn't cover other details that could potentially be changed by unrelated LLVM patches.  Otherwise, we generally aim to make IR-generation tests "unit" tests that specifically test the output of Clang's IR-generation.  For output that's only included in optimized modes, like TBAA metadata, that generally requires enabling optimization (with `-O<n>`) and then disabling all the LLVM optimizations (with `-disable-llvm-passes`).

Reply:
I am sure that level 3 optimization should cut out all stack manipulation here leaving only one necessary load(const int *x = *p) and one necessary store(*q = 0). This test checks only presence of that load/store and metadata information, so I don't think it could be disrupted by unrelated patches.